src/interfaces/graphql/resolvers/admin/PageMutation.ts

```diff
@@ -0,0 +1,32 @@
+import { GraphQLError } from 'graphql';
+
+import type { GraphQLContext } from '~/back-shared/types/container/types';
+import { graphqlErrors } from '~/constants/errors';
+import { PageService } from '~/domain/services/pageService';
+import { PageRepository } from '~/infrastructure/repositories/pageRepository/pageRepository';
+import type { Page, Scalars } from '~/types/graphql/generated/graphql';
+
+type UpdatePageBlocksArgs = {
+  input: {
+    slug: string;
+    blocks: Scalars['JSON']['input'];
+  };
+};
+
+export const PageMutation = {
+  async updatePageBlocks(_: unknown, { input }: UpdatePageBlocksArgs, context: GraphQLContext): Promise<Page> {
+    if (!context.admin) {
+      throw new GraphQLError(graphqlErrors.UNAUTHENTICATED.message, {
+        extensions: { code: graphqlErrors.UNAUTHENTICATED.code }
+      });
+    }
+
+    const service = PageService(PageRepository());
```

Comment:
If you add this service to DI container so you will can get it from a `context`
```suggestion
    const service = context.requestContainer.resolve('PageService');
```